arch/arm/arm-m/thread.c

```diff
@@ -118,10 +121,48 @@ __NAKED void _pendsv(void)
         "pop	{ r4-r11, lr };"
         "bx		lr;"
         :: "i" (pendsv)
+        );
+    __UNREACHABLE;
+#else
+    struct arm_cm_exception_frame_long *frame;
+
+    __asm__ volatile(
```

Comment:
I'd be very careful about this code. Theoretically you cannot emit any C code in a _NAKED function since there was no stack frame set up. I think this *might* be okay because the state is saved in asm, but I'm not so sure it's okay. That's the reason I had the convoluted branch to another function that did the body of this in the M3 version.